gems/aws-sdk-core/spec/shared_spec_helper.rb

```diff
@@ -15,9 +15,6 @@
 #
 RSpec.configure do |config|
   config.before(:each) do
-
-    stub_const('ENV', {})
```

Comment:
I'd prefer to keep this here rather than anyone having to remember to add stubbing to individual spec files when necessary.. this seems like it has an important job to potentially be overlooked (to prevent real credentials from being used).

I also think that here lies the actual problem.. rather than stubbing ENV as a hash (because it's not a hash - it's [hash-LIKE](https://docs.ruby-lang.org/en/master/ENV.html)) we should actually be clearing the ENV with `ENV.clear` or better if there's a way to `stub_const('ENV', somehow_make_a_new_ENV_here)`.